### PR TITLE
ERM-2045: Extend Package data model to store additional information

### PR DESCRIPTION
**Pull request by peterboehm**

### CODE REVIEW

File: service/grails-app/views/alternateResourceNames/_alternateResourceNames.gson [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import groovy.transform.*
+import org.olf.kb.AlternateResourceName
+
+@Field AlternateResourceName alternateResourceName
+json g.render(alternateResourceName)
```
Comment (EthanFreestone) [inline service/grails-app/views/alternateResourceNames/_alternateResourceNames.gson:5]: I would like these to exclude "owner". We will only ever fetch these from their owner anyway, so it seems unnecessary to return that data.

File: service/grails-app/views/packageDescriptionUrl/_packageDescriptionUrl.gson [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import groovy.transform.*
+import org.olf.kb.PackageDescriptionUrl
+
+@Field PackageDescriptionUrl packageDescriptionUrl
+json g.render(packageDescriptionUrl)
```
Comment (EthanFreestone) [inline service/grails-app/views/packageDescriptionUrl/_packageDescriptionUrl.gson:5]: I would like these to exclude "owner". We will only ever fetch these from their owner anyway, so it seems unnecessary to return that data.

File: service/grails-app/views/contentType/_contentType.gson [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import groovy.transform.*
+import org.olf.kb.ContentType
+
+@Field ContentType contentType
+json g.render(contentType)
```
Comment (EthanFreestone) [inline service/grails-app/views/contentType/_contentType.gson:5]: I would like these to exclude "owner". We will only ever fetch these from their owner anyway, so it seems unnecessary to return that data.

**Review by EthanFreestone — CHANGES_REQUESTED** [review]

File: service/grails-app/views/alternateResourceNames/_alternateResourceNames.gson [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import groovy.transform.*
+import org.olf.kb.AlternateResourceName
+
+@Field AlternateResourceName alternateResourceName
+json g.render(alternateResourceName, [excludes: ['owner']])
```
Comment (peterboehm) [inline service/grails-app/views/alternateResourceNames/_alternateResourceNames.gson:5]: this seems to be ignored while working on the other two

**Comment by peterboehm** [comment]
I pushed changes to exclude the owner on the JSON, but the exclude seems to be ignored on alternateResourceName, see comment above

File: service/grails-app/domain/org/olf/kb/ContentType.groovy [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package org.olf.kb;
+
+import grails.gorm.MultiTenant
+
+public class ContentType implements MultiTenant<ContentType> {
+
+	String id
+  String contentType
```
[the remark this answers is not in the document]
Comment (peterboehm) [reply service/grails-app/domain/org/olf/kb/ContentType.groovy]: Changed to RefData with values "Serials", "Monographs", "Databases"

File: service/grails-app/views/packageDescriptionUrl/_packageDescriptionUrl.gson (hunk 2 above)
Comment (peterboehm) [reply service/grails-app/views/packageDescriptionUrl/_packageDescriptionUrl.gson]: done

File: service/grails-app/views/contentType/_contentType.gson (hunk 3 above)
Comment (peterboehm) [reply service/grails-app/views/contentType/_contentType.gson]: done

File: service/grails-app/views/alternateResourceNames/_alternateResourceNames.gson (hunk 1 above)
Comment (peterboehm) [reply service/grails-app/views/alternateResourceNames/_alternateResourceNames.gson]: exclude on 'owner' still doesn't work

**Comment by peterboehm** [comment]
Excluding 'owner' on 'alternateResourceNames' still doesn't work after change to ermResource.gson as discussed today

**Pull request merged**